feat: Add capability to generate TAF compose files (#373)
edgexfoundry/developer-scripts: compose-builder/README.md

    Device Service **extending** compose file, which adds the **Device SNMP**  service.
- **add-device-virtual.yml**
    Device Service **extending** compose file, which adds the **Device Virtual**  service.
- **add-asc-http-export.yml**
    Application Service Configurable **extending** compose file, which adds the **App Service Http Export**  service.
- **add-asc-http-export-secure.yml**
    Application Service Configurable **extending** compose file, which adds the **App Service Http Export Secrets** service. Service is enable for secure secrets.
- **add-asc-mqtt-export.yml**
    Application Service Configurable **extending** compose file, which adds the **App Service MQTT Export**  service.
- **add-asc-MQTT-export-secure.yml**

>     Application Service Configurable **extending** compose file, which adds the **App Service Http Export Secrets** service. Service is enable for secure secrets.

Reviewer: Should **App Service Http Export Secrets** be **App Service MQTT Export Secrets**?

Author: Yes! good catch. Thanks!

Author: fixed

- **add-modbus-simulator.yml**
    ModBus Simulator **extending** compose file. Adds the MQTT ModBus Simulator service. Must be used in conjunction with  **add-device-modbus.yml**
- **add-mqtt-broker.yml**
    MQTT Broker **extending** compose file. Adds the Eclipse Mosquitto MQTT Broker.
- **add-mqtt-messagebus.yml**

>     MQTT MesssageBus **extending** compose file. Adds additional configuration of services so that the `MQTT` implementation of the Edgex Message Bus is used. Must be use in conjunction with **add-mqtt-broker.yml**

Reviewer 2: ```suggestion
    MQTT MesssageBus **extending** compose file. Adds additional configuration of services so that the `MQTT` implementation of the Edgex Message Bus is used. **Must be used in conjunction with add-mqtt-broker.yml**
```
would prefer to stress on the whole must be part

- **add-taf-app-services.yml**
    TAF App Services **extending** compose file. Adds additional App Service for the TAF testing compose files.
- **add-taf-device-services-mods.yml**

>     TAF Device Services **extending** compose file. Modifies setting of Device Virtual and Device Modbus for the TAF testing compose files. Must be used in conjunction with **add-device-modbus.yml** and **add-device-virtual.yml**

Reviewer 2: ```suggestion
    TAF Device Services **extending** compose file. Modifies setting of Device Virtual and Device Modbus for the TAF testing compose files. **Must be used in conjunction with add-device-modbus.yml and add-device-virtual.yml**
```
would prefer to stress on the whole must be part

```
```
build

> Generates the all standard Edgex compose file variationsand the TAF testing compose files. Stores them in the configured relese folder. Each variation or standard compose files, except UI, includes Device REST & Device Virtual. Compose files are named appropriatly for release and options used to generate them. TAF compose files are store in the 'taf' sub-folder

Reviewer 2: ```suggestion
Generates the all standard Edgex compose file variations and the TAF testing compose files. The generated compose files are stored in the configured release folder. Each variation or standard compose files, except UI, includes Device REST & Device Virtual. Compose files are named appropriately for release and options used to generate them. TAF compose files are store in the 'taf' sub-folder
```
some typos

    ds-rest:    Generates compose file with device-rest included
    ds-snmp:    Generates compose file with device-snmp included
    ds-virtual: Generates compose file with device-virtual included
    modbus-sim: Generates compose file with ModBus simulator included
    asc-http:   Generates compose file with App Service HTTP Export included
    asc-http-s: Generates compose file with App Service HTTP Export Secrets included
    asc-mqtt:   Generates compose file with App Service MQTT Export included
    asc-mqtt-s: Generates compose file with App Service MQTT Export Secrets included
    mqtt:       Generates compose file with services configure for MQTT Message Bus 
    ui:         Generates stand-alone compose file for EdgeX UI	
```

```
taf-compose [options] 
Generates a TAF general testing compose file as specified by options and stores them in the configured TAF release folder. Compose files are named appropriatly for the options used to generate them.

Options:
    taf-secty:	  Bulds geneneral TAF testing compose file with security services
    taf-no-secty: Bulds geneneral TAF testing compose file without security services

>     no-secty: 	  Must be use in conjunction with taf-no-secty so compose file is named appropriatly

Reviewer 2: interesting. so it could not be combined into `taf-no-secty` regarding the naming?

Author: Actually, yes it could. Good call.
